src/components/SendTransaction/CreateTransaction.tsx

```diff
@@ -212,50 +213,55 @@ export const CreateTransaction = ({
   }, [knownAccount, memoContent]);
 
   const fetchIfFederationAddress = async () => {
+    setFederationAddressError("");
+
     if (isFederationAddress(toAccountId)) {
       setFederationAddressFetchStatus(ActionStatus.PENDING);
 
       try {
         const response = await FederationServer.resolve(toAccountId);
+
         setFederationAddressFetchStatus(ActionStatus.SUCCESS);
+        setFederationAddress(response.account_id);
+        checkIfAccountIsFunded(response.account_id);
+
+        if (!StrKey.isValidEd25519PublicKey(response.account_id)) {
+          setFederationAddressError(
+            "Stellar address or public key resolved from federation address is invalid",
+          );
+        }
 
         if (response.memo || response.memo_type) {
           setIsMemoVisible(true);
-
-          if (response.memo_type) {
-            setIsMemoTypeFromFederation(true);
-          }
-
-          if (response.memo) {
-            setIsMemoContentFromFederation(true);
-          }
-
-          setFederationAddress(response.account_id);
           setMemoType(response.memo_type || StellarSdk.MemoText);
           setMemoContent(response.memo || "");
-        } else {
-          setFederationAddress(response.account_id);
+          setIsMemoTypeFromFederation(Boolean(response.memo_type));
+          setIsMemoContentFromFederation(Boolean(response.memo));
+        } else if (knownAccounts[response.account_id]) {
+          setIsMemoVisible(true);
+          setMemoType(StellarSdk.MemoText);
+          setMemoContent(response.memo || "");
         }
       } catch (err) {
-        setFederationAddressFetchStatus(ActionStatus.ERROR);
+        setFederationAddressError("Federation Address not found");
+        setFederationAddressFetchStatus(null);
       }
     } else {
       resetFederationAddressInput();
     }
   };
 
-  const checkIfAccountIsFunded = async () => {
-    setIsCheckingAddress(true);
-
-    if (!toAccountId || !StrKey.isValidEd25519PublicKey(toAccountId)) {
+  const checkIfAccountIsFunded = async (accountId: string) => {
```

Comment:
```suggestion
  const checkAndSetIsAccountFunded = async (accountId: string) => {
```

Reply:
Another option could be `setIsAccountFunded(!accountId || !StrKey.isValidEd25519PublicKey(accountId)`, the wrapper method name is kind of ambiguous what it's doing imo (it checks, and then what)

Reply:
Good point, `checkAndSetIsAccountFunded` is more clear. `setIsAccountFunded` is already taken for updating the state value.